In [5]:
%load_ext sql


In [6]:
from db_connection import NEXASAT
%sql $NEXASAT


In [7]:
%config SqlMagic.style = '_DEPRECATED_DEFAULT'

# Data Exploration 




In [8]:
# Preview of the Data  

%sql SELECT * FROM nexasat LIMIT 5;

 * postgresql://postgres:***@localhost:5432/postgres
5 rows affected.


[('NSat2001', 'Female', 'Yes', 'No', 0, 156.0, 8.36, 'Prepaid', 'Premium', 126.0, 44, 'Yes', 'Yes', 0),
 ('NSat2002', 'Male', 'No', 'No', 0, 383.0, 5.35, 'Prepaid', 'Basic', 80.0, 7, 'No', 'No', 1),
 ('NSat2003', 'Male', 'No', 'No', 0, 407.0, 3.04, 'Postpaid', 'Premium', 314.8, 15, 'Yes', 'Yes', 1),
 ('NSat2004', 'Male', 'No', 'No', 0, 106.0, 1.2, 'Prepaid', 'Premium', 126.0, 38, 'Yes', 'Yes', 0),
 ('NSat2005', 'Female', 'No', 'No', 0, 124.0, 5.63, 'Postpaid', 'Premium', 97.95, 17, 'Yes', 'Yes', 1)]

In [9]:
# Checking for missings data
%sql SELECT COUNT(*) - COUNT(Customer_ID) AS missing_data FROM nexasat;

 * postgresql://postgres:***@localhost:5432/postgres
1 rows affected.


[(0,)]

The result shows **0 missing values**, confirming that the dataset is complete.



In [10]:
# checking for duplicates
%sql SELECT Customer_ID, COUNT(*) FROM nexasat GROUP BY Customer_ID HAVING COUNT(*) > 1;


 * postgresql://postgres:***@localhost:5432/postgres
0 rows affected.


[]

##### Duplicates
The query results indicate that there are **no** duplicate records in the dataset, as no Customer_ID appears more than once. 

In [26]:


%%sql 
SELECT ROUND(AVG(Monthly_Bill_Amount::INT), 2) AS avg_bill, 
MIN(Monthly_Bill_Amount) AS min_bill, MAX(Monthly_Bill_Amount) AS max_bill, 
ROUND(AVG(Call_Duration::INT), 2) AS avg_call_duration, 
ROUND(AVG(Data_Usage::INT), 2) AS avg_data_usage, 
COUNT(DISTINCT Plan_Type) AS num_plans FROM nexasat;


 * postgresql://postgres:***@localhost:5432/postgres
1 rows affected.


[(Decimal('149.79'), 22.0, 415.0, Decimal('240.50'), Decimal('8.79'), 2)]

##### Summary:
- **Monthly Bill**: The average is 149.79, ranging from 22.00 to 415.00, showing significant variation.
- **Call Duration**: The average is 240.50 minutes, suggesting frequent or long calls.
- **Data Usage**: The average is 8.79 units, indicating low data consumption.
- **Plan Diversity**: Two distinct plans are available, suggesting limited plan options.


In [31]:


%%sql
SELECT ROUND(AVG(Tenure_Months::INT),2) AS avg_tenure, Plan_Level 
FROM nexasat 
GROUP BY Plan_Level;


 * postgresql://postgres:***@localhost:5432/postgres
2 rows affected.


[(Decimal('32.16'), 'Premium'), (Decimal('16.61'), 'Basic')]

##### Customer Tenure by Plan Level  

The results indicate that customers on the **Premium** plan have an average tenure of **32.16 months**, while those on the **Basic** plan have a shorter average tenure of **16.61 months**. This suggests that Premium subscribers tend to stay longer, possibly due to better service offerings or customer satisfaction.


In [33]:
%%sql
SELECT gender, COUNT(*) FROM nexasat GROUP BY gender;

 * postgresql://postgres:***@localhost:5432/postgres
2 rows affected.


[('Female', 3488), ('Male', 3555)]

##### Gender Distribution  

The dataset has a nearly equal distribution of genders, with **3,488 females** and **3,555 males**. This balance suggests that the customer base is fairly evenly split between male and female users.


In [35]:
%%sql
SELECT Partner, COUNT(*) FROM nexasat GROUP BY Partner; 


 * postgresql://postgres:***@localhost:5432/postgres
2 rows affected.


[('No', 3641), ('Yes', 3402)]

##### Partner Status Distribution  

The dataset shows that **3,641 customers do not have a partner**, while **3,402 customers do**. This indicates a relatively balanced distribution between the two groups.


In [36]:
%%sql
SELECT Dependents, COUNT(*) FROM nexasat GROUP BY Dependents;

 * postgresql://postgres:***@localhost:5432/postgres
2 rows affected.


[('No', 4933), ('Yes', 2110)]

##### Dependents Distribution  

The majority of customers **(4,933)** do not have dependents, while **2,110** do. This suggests that a significant portion of the customer base may be independent individuals, which could influence service preferences and spending patterns.


In [44]:
%%sql 
SELECT Plan_Level, 
ROUND(AVG(Tenure_Months::INT),2) AS average_tenure, 
COUNT(Customer_ID) AS number_of_users
FROM nexasat WHERE Churn = 1
GROUP BY 1;

 * postgresql://postgres:***@localhost:5432/postgres
2 rows affected.


[('Premium', Decimal('26.08'), 565), ('Basic', Decimal('18.55'), 2206)]

##### Churn and Tenure by Plan Level  


The **Premium plan**,  had **565** churned users with an average tenure of **26.08 months**, while the **Basic plan** had **2,206** churned users with a shorter average tenure of **18.55 months**.  

Since **Premium users are locked in for a minimum of 12 months**, their longer tenure may reflect a commitment period and potentially better service. However, despite staying longer, some still eventually churn, while Basic users, who have no such lock-in, churn in higher numbers and at a faster rate.  


# Creating a Table for Existing Users and Exploring the Dataset

To better analyze the behavior of our retained customers, we first created a table that includes only users who have not churned, referred to as **existing users**. This allows us to focus on active customers, providing a more accurate view of user behavior and engagement. 

In this step, we will explore the dataset to gain insights into key metrics such as revenue and usage patterns for these existing users. 


In [45]:
#Create a table of existing users only

%%sql
CREATE TABLE existing_users AS
SELECT * FROM nexasat
WHERE Churn = 0;



 * postgresql://postgres:***@localhost:5432/postgres
4272 rows affected.


[]

##### Existing Users Segmentation  

A total of **4,272** users have been classified as existing users (i.e., non-churned), making up a significant portion of the **7,043** total users. This indicates that approximately **39.3% of users have churned**, highlighting potential areas for retention strategies.  


In [58]:
%%sql
SELECT Plan_Level, COUNT(Customer_ID) AS total_users, ROUND(AVG(Tenure_Months::INT),2)
FROM existing_users 
GROUP BY 1;

 * postgresql://postgres:***@localhost:5432/postgres
2 rows affected.


[('Premium', 3015, Decimal('33.30')), ('Basic', 1257, Decimal('13.21'))]



##### Distribution of Existing Users by Plan Level  

Among the **4,272 retained users**, the majority (**3,015 users**) are on the **Premium plan**, with an **average tenure of 33.3 months**, while **1,257 users** remain on the **Basic plan**, averaging **13.21 months**. This suggests that **Premium users tend to stay longer**, possibly due to added benefits such as **tech support and multiple lines**. In contrast, **Basic plan users have a significantly shorter tenure**, highlighting an opportunity for targeted upgrades to **enhance retention and reduce future churn**.  


In [57]:
%%sql
SELECT Plan_Level , ROUND(SUM(Monthly_Bill_Amount::INT),2) AS Revenue
FROM existing_users 
GROUP BY 1;


 * postgresql://postgres:***@localhost:5432/postgres
2 rows affected.


[('Premium', Decimal('522971.00')), ('Basic', Decimal('150026.00'))]

##### Revenue Contribution by Plan Level  

Premium users generate a significantly higher revenue of **522,971**, compared to  **150,026** from Basic users. This reinforces the idea that **Premium customers are a key revenue driver**, likely due to higher service costs, multiple lines, and added tech support. Encouraging Basic users to upgrade could further boost overall revenue while enhancing customer retention.  


In [49]:

%%sql

SELECT COUNT(Customer_ID) FROM existing_users
WHERE Plan_level = 'Basic' AND (Dependents = 'Yes' OR Partner = 'Yes');


 * postgresql://postgres:***@localhost:5432/postgres
1 rows affected.


[(696,)]

##### Identifying Potential Upgraders  

Among the existing users, **696 Basic plan customers** have either a partner or dependents. These users may benefit from upgrading to the **Premium package**, which offers **tech support and multiple lines**, potentially improving their overall experience and reducing churn risk. Targeted promotions or tailored incentives could encourage migration to the Premium plan.  


# Customer Lifetime Value (CLV) Analysis

In this section, we will calculate and interpret the **Customer Lifetime Value (CLV)** for different user segments. CLV is a critical metric that helps understand the long-term value of customers and can be used to optimize marketing, sales, and customer retention strategies. By analyzing the **average revenue per user** and the **tenure of users**, we can gain better insights into the profitability and sustainability of the existing customer base.


In [68]:
#Alter the table and adding Customer Life value, Clv column
%%sql
ALTER TABLE existing_users ADD COLUMN Clv FLOAT; 

 * postgresql://postgres:***@localhost:5432/postgres
Done.


[]

In [69]:
%%sql
UPDATE existing_users 
SET Clv = (Monthly_Bill_Amount * Tenure_Months);



 * postgresql://postgres:***@localhost:5432/postgres
4272 rows affected.


[]

##### Calculating Customer Lifetime Value (CLV)

The **Customer Lifetime Value (CLV)** is calculated by multiplying the **Monthly Bill Amount** by the **Tenure in Months** for each customer. This provides an estimate of the total revenue a customer is expected to generate over the course of their relationship with the company. The CLV value is updated in the **existing_users** table to reflect this calculation for each customer.


In [70]:
%%sql
SELECT Customer_ID, Clv
From existing_users LIMIT 5;

 * postgresql://postgres:***@localhost:5432/postgres
5 rows affected.


[('NSat4220', 720.0),
 ('NSat6095', 264.0),
 ('NSat8675', 4788.0),
 ('NSat2001', 5544.0),
 ('NSat2004', 4788.0)]

##### Displaying Clv  
This table shows the **Customer_ID** along with their corresponding **Customer Lifetime Value (CLV)**.

In [73]:
# Alter the table and adding Customer Life value, Clv column
%sql ALTER TABLE existing_users ADD COLUMN Clv_Score FLOAT;

 * postgresql://postgres:***@localhost:5432/postgres
Done.


[]

##### Customer Lifetime Value (CLV) Score Calculation

The **Customer Lifetime Value (CLV)** score is a critical metric that helps predict the total revenue a customer is expected to bring to the company over the course of their relationship. CLV is calculated by multiplying a customer's **Monthly Bill Amount** by their **Tenure in Months**.

In this analysis, **CLV scores** were calculated using a **Random Forest model**, which takes into account various customer features to predict their lifetime value. The model was trained on multiple factors, including:

- **Monthly Bill Amount**
- **Tenure (in months)**
- **Call Duration**
- **Data Usage**
- **Plan Level**
- **Partner/Dependents Status**
- **Senior Citizen Status**

The **percentage weight** of each feature in determining the CLV score is as follows:

| Feature                   | Percentage Weight |
|---------------------------|-------------------|
| Data_Usage                 | 23.12%            |
| Tenure_Months              | 19.40%            |
| Call_Duration              | 18.68%            |
| Plan_Level_Premium         | 16.64%            |
| Monthly_Bill_Amount        | 15.51%            |
| Plan_Type_Prepaid          | 2.11%             |
| Partner_Yes                | 1.73%             |
| Dependents_Yes             | 1.42%             |
| Senior_Citizen             | 1.40%             |

This breakdown highlights which features most influence the CLV calculation, with **Data Usage** and **Tenure** being the strongest predictors. The results and deeper analysis can be found in the **percentage-weight-clv.ipynb** file, where you can explore the full model and gain further insights.


In [75]:


%%sql 
UPDATE existing_users
SET Clv_Score = 
    (Data_Usage * 0.2336) +
    (Tenure_Months * 0.1960) +
    (Call_Duration * 0.1886) +
    (Monthly_Bill_Amount * 0.1566) +
    (CASE WHEN Plan_Level = 'Premium' THEN 1 ELSE 0 END * 0.1680) +
    (CASE WHEN Plan_Type = 'Prepaid' THEN 1 ELSE 0 END * 0.0213) +
    (CASE WHEN Partner = 'Yes' THEN 1 ELSE 0 END * 0.0174) +
    (CASE WHEN Dependents = 'Yes' THEN 1 ELSE 0 END * 0.0142) +
    (Senior_Citizen * 0.0140);


 * postgresql://postgres:***@localhost:5432/postgres
4272 rows affected.


[]

In [79]:
%%sql
SELECT 
    ROUND(MIN(Clv_Score::INT),2) AS Min_CLV_Score,
    ROUND(AVG(Clv_Score::INT),2) AS Avg_CLV_Score,
    ROUND(MAX(Clv_Score::INT),2) AS Max_CLV_Score
    
FROM existing_users;




 * postgresql://postgres:***@localhost:5432/postgres
1 rows affected.


[(Decimal('8.00'), Decimal('80.03'), Decimal('164.00'))]

##### CLV Score Analysis

The **Customer Lifetime Value (CLV) scores** for the existing users range from a **minimum value of 8.00** to a **maximum value of 164.00**, with an **average score of 80.03**. 

This indicates a significant variation in the lifetime value of customers. The lower CLV scores may represent customers with shorter tenures or lower monthly bills, while the higher scores likely reflect long-term, high-value customers. The average CLV score of 80.03 suggests a balanced customer base, with room for targeting high-value customers for retention strategies and low-value customers for potential upgrades or retention efforts.


##### Customer Segmentation Based on CLV Score

The **Customer Lifetime Value (CLV) scores** are used to classify customers into different segments based on their value to the company. This segmentation is performed using the 25th, 50th, and 75th percentiles to divide the customers into four distinct categories:


In [80]:

%%sql
ALTER TABLE existing_users ADD COLUMN Clv_Segment VARCHAR(20);

 * postgresql://postgres:***@localhost:5432/postgres
Done.


[]

In [81]:
%%sql
UPDATE existing_users
SET Clv_Segment =
    CASE
        WHEN Clv_Score > (SELECT PERCENTILE_CONT(0.75) WITHIN GROUP (ORDER BY Clv_Score) FROM existing_users)
             THEN 'High Value'
        WHEN Clv_Score > (SELECT PERCENTILE_CONT(0.50) WITHIN GROUP (ORDER BY Clv_Score) FROM existing_users)
             THEN 'Moderate Value'
        WHEN Clv_Score > (SELECT PERCENTILE_CONT(0.25) WITHIN GROUP (ORDER BY Clv_Score) FROM existing_users)
             THEN 'Low Value'
        ELSE 'Churn Risk'
    END;

 * postgresql://postgres:***@localhost:5432/postgres
4272 rows affected.


[]


##### Customer Segmentation
1. **High Value**: 
   Customers with CLV scores above the 75th percentile (Top 25% of customers). These are the most valuable customers, contributing significantly to the company’s revenue. They are likely to be long-term, high-paying customers who may benefit from personalized offers and loyalty programs.

2. **Moderate Value**: 
   Customers with CLV scores between the 50th and 75th percentiles. These customers are considered average or mid-value. They contribute a steady amount to the company but may need more targeted efforts to encourage retention and increase their lifetime value.

3. **Low Value**: 
   Customers with CLV scores between the 25th and 50th percentiles. These customers are in the lower tier, and although they bring in some revenue, there may be opportunities to enhance their value, either through upgrades or targeted engagement strategies.

4. **Churn Risk**: 
   Customers whose CLV scores are below the 25th percentile. These customers are at a higher risk of churn due to their low lifetime value. Efforts should focus on identifying pain points and encouraging these customers to upgrade or renew their commitment to the service.

This segmentation helps the company target different customer groups effectively, with tailored strategies for retention, engagement, and growth.

In [99]:
# Displaying Customers with their respective Clv, Clv score and Clv segment

%%sql
SELECT Customer_ID, Clv, ROUND(Clv_score::INT,2) AS Clv_score, Clv_segment from existing_users LIMIT 20;

 * postgresql://postgres:***@localhost:5432/postgres
20 rows affected.


[('NSat2117', 2142.0, Decimal('83.00'), 'Moderate Value'),
 ('NSat2121', 6426.0, Decimal('97.00'), 'Moderate Value'),
 ('NSat2123', 5796.0, Decimal('99.00'), 'Moderate Value'),
 ('NSat2124', 3024.0, Decimal('104.00'), 'Moderate Value'),
 ('NSat2127', 5166.0, Decimal('83.00'), 'Moderate Value'),
 ('NSat2128', 6552.0, Decimal('56.00'), 'Low Value'),
 ('NSat2129', 6174.0, Decimal('91.00'), 'Moderate Value'),
 ('NSat2131', 6930.0, Decimal('108.00'), 'High Value'),
 ('NSat2133', 6678.0, Decimal('80.00'), 'Moderate Value'),
 ('NSat2143', 7560.0, Decimal('117.00'), 'High Value'),
 ('NSat2149', 2142.0, Decimal('90.00'), 'Moderate Value'),
 ('NSat2151', 3402.0, Decimal('120.00'), 'High Value'),
 ('NSat2156', 6048.0, Decimal('122.00'), 'High Value'),
 ('NSat2157', 6174.0, Decimal('98.00'), 'Moderate Value'),
 ('NSat2166', 6552.0, Decimal('63.00'), 'Low Value'),
 ('NSat2172', 7434.0, Decimal('121.00'), 'High Value'),
 ('NSat2179', 756.0, Decimal('54.00'), 'Low Value'),
 ('NSat2184', 3906.0, Decimal('60.00'), 'Low Value'),
 ('NSat2192', 4410.0, Decimal('47.00'), 'Churn Risk'),
 ('NSat2193', 5670.0, Decimal('123.00'), 'High Value')]

##### Analyzing the Customer Segments

In this section, we will analyze the different customer segments based on their **Customer Lifetime Value (CLV)** scores. By segmenting customers into **High Value**, **Moderate Value**, **Low Value**, and **Churn Risk** categories, we can better understand their behaviors, needs, and potential opportunities for engagement.


In [118]:
%%sql
SELECT Clv_segment,
       ROUND(AVG(Monthly_bill_amount::INT),2) AS Avg_bill, 
       ROUND(SUM(Monthly_Bill_Amount::INT),2)AS Total_Revenue,
       ROUND(AVG(Tenure_Months::INT),2) AS Avg_tenure 
FROM existing_users
GROUP BY Clv_segment;


 * postgresql://postgres:***@localhost:5432/postgres
4 rows affected.


[('Churn Risk', Decimal('108.09'), Decimal('115440.00'), Decimal('20.99')),
 ('Low Value', Decimal('132.45'), Decimal('141458.00'), Decimal('27.61')),
 ('Moderate Value', Decimal('153.35'), Decimal('163777.00'), Decimal('28.05')),
 ('High Value', Decimal('236.26'), Decimal('252322.00'), Decimal('32.90'))]

##### Interpreting CLV Segments  

The **Customer Lifetime Value (CLV) segmentation** reveals key differences in **average monthly bill** and **average tenure** across customer groups:

- **Churn Risk** customers have the lowest **average bill ($108.09)** and shortest **tenure (20.99 months)**, indicating  lower  engagement  and  a higher  likelihood  of  leaving. Addressing their concerns through targeted retention efforts could help reduce churn.
- **Low Value** customers show a moderate increase in **average bill ($132.45)** and **tenure (27.61 months)**, suggesting they stay longer but still have room for increased spending or engagement.
- **Moderate Value** customers exhibit a **higher bill ($153.35)** with a slightly longer **tenure (28.05 months)**, indicating stronger retention and spending habits.
- **High Value** customers contribute the most, with a **significantly higher average bill ($236.26)** and the longest **tenure (32.90 months)**. These loyal customers may benefit from exclusive offers or loyalty programs to maintain their engagement.

This analysis suggests that increasing customer engagement and retention efforts, particularly for **Churn Risk** and **Low Value** customers, could help enhance overall revenue and reduce churn.


In [116]:
%%sql
SELECT Clv_segment,
COUNT(plan_level) as Premium_users FROM existing_users
WHERE plan_level = 'Premium'
GROUP BY 1;

 * postgresql://postgres:***@localhost:5432/postgres
4 rows affected.


[('Churn Risk', 508),
 ('Low Value', 760),
 ('Moderate Value', 793),
 ('High Value', 954)]

##### Premium Users Across CLV Segments  

The distribution of **Premium users** across CLV segments highlights key customer trends:  

- **High Value customers (954 users)** have the most **Premium plan users**, showing their long-term engagement and higher spending.  
- **Moderate Value (793 users)** and **Low Value (760 users)** segments also have a significant number of Premium users, suggesting potential for increased value through upselling or retention strategies.  
- **Churn Risk (508 users)** includes a notable share of Premium users, indicating that even higher-tier customers are at risk of leaving. Targeted interventions, such as personalized support or incentives, could help retain them.  

This insight can guide strategies to maximize customer lifetime value and reduce churn.  


In [120]:
%%sql 
SELECT 
    Clv_Segment,
    SUM(CASE WHEN Senior_Citizen = 1 THEN 1 ELSE 0 END) AS senior_citizens,
    SUM(CASE WHEN Partner = 'Yes' THEN 1 ELSE 0 END) AS partnered_customers,
    SUM(CASE WHEN Dependents = 'Yes' THEN 1 ELSE 0 END) AS customers_with_dependents
FROM existing_users
GROUP BY Clv_Segment
ORDER BY 
    CASE Clv_Segment
        WHEN 'High Value' THEN 1
        WHEN 'Moderate Value' THEN 2
        WHEN 'Low Value' THEN 3
        WHEN 'Churn Risk' THEN 4
    END;


 * postgresql://postgres:***@localhost:5432/postgres
4 rows affected.


[('High Value', 158, 510, 329),
 ('Moderate Value', 158, 523, 332),
 ('Low Value', 189, 537, 281),
 ('Churn Risk', 168, 513, 342)]

##### **Customer Segments Analysis: Senior Citizens, Partnered Customers, and Dependents**  

This reveals key trends:  

- **Senior Citizens:** Their numbers remain fairly stable across segments (158–189). However, a **slight increase in the "Low Value" and "Churn Risk" groups** suggests they may struggle with service usage, possibly due to tech challenges.  

- **Partnered Customers:** The distribution is **even across all segments**, indicating that having a partner does not significantly impact customer value. Other factors like **plan type and service usage** play a bigger role.  

- **Customers with Dependents:** The **"Churn Risk" segment has the most dependents (342),** while the **"Low Value" group has the least (281).** This may indicate financial constraints or lower engagement, making them more likely to churn.  

These insights highlight opportunities to **target senior customers with tech support** and explore **family-friendly offers** for customers with dependents.  


In [128]:
%%sql
SELECT Clv_Segment, 
       Plan_LEVEL, 
       COUNT(Customer_ID) AS total_users
FROM existing_users
GROUP BY Clv_Segment, Plan_LEVEL
ORDER BY 
    CASE Clv_Segment
        WHEN 'High Value' THEN 1
        WHEN 'Moderate Value' THEN 2
        WHEN 'Low Value' THEN 3
        WHEN 'Churn Risk' THEN 4
    END;


 * postgresql://postgres:***@localhost:5432/postgres
8 rows affected.


[('High Value', 'Basic', 114),
 ('High Value', 'Premium', 954),
 ('Moderate Value', 'Premium', 793),
 ('Moderate Value', 'Basic', 275),
 ('Low Value', 'Premium', 760),
 ('Low Value', 'Basic', 308),
 ('Churn Risk', 'Premium', 508),
 ('Churn Risk', 'Basic', 560)]

##### CLV Segment by Plan Level Summary

- **High Value**: Predominantly **Premium** users, highlighting the role of premium services in high value.
- **Moderate Value**: More **Premium** users than **Basic**, suggesting premium services are linked to moderate value.
- **Low Value**: Despite more **Premium** users, their value may be low due to underutilization. 
- **Churn Risk**: Higher number of **Basic** users, indicating **Basic** plans are more likely to face churn. 

**Premium** plans tend to correlate with higher customer value, while **Basic** plans show higher churn risk and lower lifetime value.
This highlights the importance of **premium offerings** in enhancing customer retention and overall lifetime value.


In [129]:
%%sql
SELECT 
    Clv_Segment,
    ROUND(AVG(Data_Usage::INT), 2) AS avg_data_usage,
    ROUND(AVG(Call_Duration::INT), 2) AS avg_call_duration
FROM existing_users
GROUP BY Clv_Segment;


 * postgresql://postgres:***@localhost:5432/postgres
4 rows affected.


[('Churn Risk', Decimal('7.60'), Decimal('89.40')),
 ('Low Value', Decimal('8.55'), Decimal('197.16')),
 ('Moderate Value', Decimal('8.67'), Decimal('311.83')),
 ('High Value', Decimal('7.96'), Decimal('418.17'))]

##### CLV Segment: Data Usage & Call Duration

- **Churn Risk**: Low usage (7.60) and call duration (89.40), minimal engagement.
- **Low Value**: Moderate usage (8.55) and call duration (197.16), potential for growth.
- **Moderate Value**: Higher usage (8.67) and call duration (311.83), good engagement with room for improvement.
- **High Value**: High call duration (418.17) with moderate data usage (7.96), strong engagement due to premium features.

These insights suggest that promoting data and call usage to lower-value segments could help drive engagement and potentially elevate their customer value.

# Marketing Strategies Based on CLV Analysis

In this section, we will recommend targeted marketing strategies aimed at improving customer engagement and boosting customer lifetime value (CLV) across different segments. The strategies will be based on insights derived from the CLV analysis and customer behavior patterns identified.


In [131]:
%%sql
SELECT COUNT(Customer_ID)
FROM existing_users
WHERE senior_citizen = 1 -- senior citizens
AND dependents = 'No' -- No children or tech savvy helpers
AND tech_support = 'No'  -- do not already have this service
AND (Clv_segment = 'Churn Risk' OR Clv_segment = 'Low Value');

 * postgresql://postgres:***@localhost:5432/postgres
1 rows affected.


[(88,)]

##### Cross-Selling Tech Support to Senior Citizens

Based on the analysis, we identified **88 senior citizens** with **no dependents** and **no tech support**, falling within the **Churn Risk** or **Low Value** segments. These customers may struggle to fully utilize the services, possibly due to lack of tech-savvy assistance. 

A targeted marketing strategy offering **tech support** as an add-on could help improve their experience, increase engagement, and potentially boost their CLV. Cross-selling tech support to this group can enhance retention and mitigate churn risk.


In [135]:
%%sql
SELECT count(customer_id) as no_of_users
FROM existing_users
WHERE multiple_line = 'No'
AND (dependents = 'Yes' OR Partner = 'Yes')
AND plan_level = 'Basic';

 * postgresql://postgres:***@localhost:5432/postgres
1 rows affected.


[(376,)]

##### Cross-Selling Multiple Lines to Customers with Dependents or Partners

We identified **376 users** on the **Basic plan** who have **dependents** or a **partner** but are not currently subscribed to **multiple lines**. These customers could benefit from the additional flexibility that multiple lines offer.

A targeted campaign promoting the **multiple lines** feature could enhance their experience, increase usage, and drive additional revenue, potentially improving their customer lifetime value.


In [140]:
%%sql
SELECT customer_id, monthly_bill_amount
FROM existing_users
WHERE plan_level = 'Basic'
AND (Clv_segment = 'High Value' OR Clv_segment = 'Moderate Value')
AND monthly_bill_amount > 150;


 * postgresql://postgres:***@localhost:5432/postgres
137 rows affected.


[('NSat4230', 298.0),
 ('NSat4260', 318.0),
 ('NSat4317', 294.0),
 ('NSat4339', 222.0),
 ('NSat4349', 369.0),
 ('NSat4364', 309.0),
 ('NSat4378', 366.0),
 ('NSat4379', 262.0),
 ('NSat4387', 324.0),
 ('NSat4399', 255.0),
 ('NSat4448', 285.0),
 ('NSat4470', 296.0),
 ('NSat4500', 301.0),
 ('NSat4525', 348.0),
 ('NSat4587', 214.0),
 ('NSat4669', 236.0),
 ('NSat4707', 308.0),
 ('NSat4717', 225.0),
 ('NSat4764', 341.0),
 ('NSat4772', 335.0),
 ('NSat4779', 308.0),
 ('NSat4799', 361.0),
 ('NSat4847', 321.0),
 ('NSat4850', 245.0),
 ('NSat4859', 253.0),
 ('NSat4879', 253.0),
 ('NSat4905', 358.0),
 ('NSat4915', 277.0),
 ('NSat4959', 239.0),
 ('NSat5023', 346.0),
 ('NSat5074', 283.0),
 ('NSat5135', 301.0),
 ('NSat5152', 248.0),
 ('NSat5173', 235.0),
 ('NSat5184', 342.0),
 ('NSat5187', 226.0),
 ('NSat5188', 214.0),
 ('NSat5207', 340.0),
 ('NSat5219', 288.0),
 ('NSat5224', 255.0),
 ('NSat5235', 247.0),
 ('NSat5243', 298.0),
 ('NSat5284', 269.0),
 ('NSat5313', 223.0),
 ('NSat5314', 319.0),
 ('NSat5315', 243.0),
 ('NSat5371', 331.0),
 ('NSat5375', 346.0),
 ('NSat5557', 315.0),
 ('NSat5561', 275.0),
 ('NSat5571', 320.0),
 ('NSat5573', 273.0),
 ('NSat5599', 304.0),
 ('NSat5690', 308.0),
 ('NSat5691', 366.0),
 ('NSat5711', 288.0),
 ('NSat5715', 208.0),
 ('NSat5836', 264.0),
 ('NSat5846', 242.0),
 ('NSat5852', 318.0),
 ('NSat5868', 304.0),
 ('NSat6041', 238.0),
 ('NSat6053', 327.0),
 ('NSat6055', 220.0),
 ('NSat6069', 280.0),
 ('NSat6082', 282.0),
 ('NSat6091', 244.0),
 ('NSat6110', 241.0),
 ('NSat6117', 277.0),
 ('NSat6119', 313.0),
 ('NSat6124', 344.0),
 ('NSat6127', 325.0),
 ('NSat6137', 299.0),
 ('NSat6149', 289.0),
 ('NSat6162', 329.0),
 ('NSat6170', 239.0),
 ('NSat6188', 254.0),
 ('NSat6636', 401.0),
 ('NSat6637', 264.0),
 ('NSat6647', 323.0),
 ('NSat6652', 299.0),
 ('NSat6659', 266.0),
 ('NSat6661', 204.0),
 ('NSat6663', 311.0),
 ('NSat6667', 287.0),
 ('NSat6669', 355.0),
 ('NSat6677', 376.0),
 ('NSat6688', 264.0),
 ('NSat6696', 225.0),
 ('NSat6698', 389.0),
 ('NSat6706', 342.0),
 ('NSat6722', 356.0),
 ('NSat6724', 222.0),
 ('NSat6741', 269.0),
 ('NSat6743', 283.0),
 ('NSat6749', 255.0),
 ('NSat6770', 276.0),
 ('NSat6772', 338.0),
 ('NSat6828', 351.0),
 ('NSat6834', 311.0),
 ('NSat6875', 370.0),
 ('NSat6881', 232.0),
 ('NSat6887', 222.0),
 ('NSat6916', 295.0),
 ('NSat6919', 329.0),
 ('NSat6928', 208.0),
 ('NSat6953', 241.0),
 ('NSat6956', 375.0),
 ('NSat6957', 214.0),
 ('NSat6958', 220.0),
 ('NSat6970', 336.0),
 ('NSat6978', 235.0),
 ('NSat6985', 350.0),
 ('NSat6998', 331.0),
 ('NSat7013', 346.0),
 ('NSat7021', 282.0),
 ('NSat7027', 276.0),
 ('NSat7031', 381.0),
 ('NSat7053', 328.0),
 ('NSat7067', 344.0),
 ('NSat7074', 263.0),
 ('NSat7075', 259.0),
 ('NSat7078', 316.0),
 ('NSat7090', 331.0),
 ('NSat7105', 239.0),
 ('NSat7108', 396.0),
 ('NSat7130', 231.0),
 ('NSat7150', 232.0),
 ('NSat7152', 351.0),
 ('NSat7163', 270.0),
 ('NSat7166', 312.0),
 ('NSat7175', 381.0),
 ('NSat7208', 317.0),
 ('NSat7229', 284.0),
 ('NSat7233', 309.0),
 ('NSat7238', 314.0),
 ('NSat7252', 258.0)]

##### Upselling Basic to Premium for High and Moderate Value Users

We identified 389 users in the **High Value** and **Moderate Value** segments on the **Basic plan** with a monthly bill amount greater than **£150**. These users are already spending a significant amount on their current plan and could benefit from the additional features offered by the **Premium plan**.

A targeted **upselling campaign** to promote the **Premium plan** could enhance their experience, increase customer satisfaction, and boost overall revenue. This could also help improve customer retention by offering more value.
